src/compiler/tsbuildPublic.ts

```diff
@@ -302,8 +303,8 @@ export function createBuilderStatusReporter(system: System, pretty?: boolean): D
     };
 }
 
-function createSolutionBuilderHostBase<T extends BuilderProgram>(system: System, createProgram: CreateProgram<T> | undefined, reportDiagnostic?: DiagnosticReporter, reportSolutionBuilderStatus?: DiagnosticReporter) {
-    const host = createProgramHost(system, createProgram) as SolutionBuilderHostBase<T>;
+function createSolutionBuilderHostBase<T extends BuilderProgram>(system: System, createProgram: CreateProgram<T> | undefined, reportDiagnostic?: DiagnosticReporter, reportSolutionBuilderStatus?: DiagnosticReporter, jsDocParsingMode?: JSDocParsingMode) {

```

Comment:
I think we shouldnt change these apis and nor should we add it to createProgramHost gien that its going be just property on host that can be set by API user if need arises.

Reply:
Sure, I'll drop it where I can. I was not sure where we drew the line on parameters to constructors versus setting properties after the fact.

Reply:
This one has been removed.